app/src/main/java/edu/temple/templetag/UserSettingActivity.java

```diff
@@ -1,55 +1,113 @@
         super.onCreate(savedInstanceState);
         setContentView(R.layout.settings_activity);
 
-        txt_username = getIntent().getExtras().getString("username");
-        txt_email = getIntent().getExtras().getString("email");
+        displayInfo();
 
-        TextView textView = findViewById(R.id.username);
-        textView.setText(txt_username);
+        findViewById(R.id.btn_lougout).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                FirebaseAuth.getInstance().signOut();
+                startActivity(new Intent(UserSettingActivity.this, LoginActivity.class));
+                finish();
+            }
+        });
 
-        getSupportFragmentManager()
-                .beginTransaction()
-                .replace(R.id.setting_container, new UserSettingActivity.SettingsFragment(txt_username, txt_email))
-                .commit();
         ActionBar actionBar = getSupportActionBar();
         if (actionBar != null) {
             actionBar.setDisplayHomeAsUpEnabled(true);
         }
-    }
+        //Change Username
+        change_username_textview = findViewById(R.id.change_username);
+        change_username_textview.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(UserSettingActivity.this);
+                builder.setTitle("New Username");
 
-    public static class SettingsFragment extends PreferenceFragmentCompat {
+                // Set up the input
+                final EditText input = new EditText(UserSettingActivity.this);
+                // Specify the type of input expected; this, for example, sets the input as a password, and will mask the text
+                input.setInputType(InputType.TYPE_CLASS_TEXT);
+                builder.setView(input);
 
-        private String txt_username, txt_email;
+                // Set up the buttons
+                builder.setPositiveButton("Change", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialogText = input.getText().toString();
+                        changeUsername();
+                    }
+                });
+                builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.cancel();
+                    }
+                });
+                builder.show();
+            }
+        });
 
-        public SettingsFragment(String txt_username, String txt_email){
-            this.txt_email = txt_email;
-            this.txt_username = txt_username;
+        TagListFragment fragment = (TagListFragment) getSupportFragmentManager().findFragmentByTag("taglist");
```

Comment:
Could we rename TagListFragment and all its instances to be TagRecyclerViewFragment like lab 2 to match the UML? We'll also need a TagRecyclerViewAdapter class but not sure how far you went into implementing that in this PR.